### ISSUE TITLE
Add support for noise channels

### ISSUE THREAD

**Issue by hrushikesh890** [PennyLaneAI/pennylane-qulacs#16]
Add support for noise channels like 
AmplitudeDamping
PhaseDamping
DepolarizingChannel

Equivalents to these pennylane noise channels 
qulacs.gate.AmplitudeDampingNoise
qulacs.gate.DephasingNoise
qulacs.gate.DepolarizingNoise

**Comment by josh146** [comment]
Thanks for the request @hrushikesh890! This is definitely something we would like to support in PL-Qulacs; we have recently added [various noise channels to PennyLane](https://pennylane.readthedocs.io/en/stable/introduction/operations.html#noisy-channels), so it should be relatively straightforward to add support to the plugin.

**Comment by hrushikesh890** [comment]
Hi @josh146 , I am a new user of pennylane qulacs and I would love to contribute. If its straightforward i think it should be good place to start for me to understand the codebase. Can i work on this?

**Comment by josh146** [comment]
@hrushikesh890 yes for sure 🙂 I'm not too familiar with Qulacs, but it should be sufficient to:

* Update the operation map in the plugin here: https://github.com/PennyLaneAI/pennylane-qulacs/blob/a70b880cedea39d4837266753a8aa5794a719887/pennylane_qulacs/qulacs_device.py#L79

* Adding tests to make sure that the noisy channels work from PennyLane

Feel free to make a work in progress PR, and to tag us if you have any questions

**Comment by hrushikesh890** [comment]
Sorry for the sluggish progress Was busy with exams. Will write the test cases this week and finish it

**Comment by trbromley** [comment]
Hey @hrushikesh890, just wanted to follow up and see if you are still working on this?

**Comment by hrushikesh890** [comment]
Hi @trbromley  I have finished the codes but not the test cases. I have some interesting findings.  The outputs of   qulacs.gate.AmplitudeDampingNoise on qulacs and  AmplitudeDamping when run on mixed simulator, do not match. Somehow the output of qulacs is a vector which I find very confusing. I am still looking into this and trying to understand where I am wrong

**Comment by trbromley** [comment]
Thanks @hrushikesh890! Does [this](http://docs.qulacs.org/en/latest/guide/2.0_python_advanced.html#CPTP-mapping) help? It looks like `gate.update_quantum_state(state)` is the way to go.

**Comment by davzuliani** [comment]
Hi @hrushikesh890! I'm also interested in this topic since I'm trying to find a way to implement a noisy circuit for my application (QML for HEP). Is there any news about this?

**Comment by antalszava** [comment]
Hi everyone, just a fly-by thought: could the mismatch be due to the difference in qubit ordering that PennyLane and Qulacs use?

**Comment by hrushikesh890** [comment]
Hi I tried gate.update_quantum_state(state) however it doesnt seems to help. It seems to me that Qulacs noise gates do not use a density  matrix simulation

**Comment by co9olguy** [comment]
I just had a look at qulacs source code. It could be that their noisy simulation is "monte carlo" style, i.e., it projects stochastically onto one of the possible output states (e.g., flip a bit), rather than using CPTP channels on density matrices like PennyLane's mixed simulator. If that's the case, we won't be able to support it at the moment (until we introduce support for this kind of probabilistic simulator device)
